app/scheduled/circulars/circularAction.ts

```diff
@@ -0,0 +1,22 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { Circular } from '~/routes/circulars/circulars.lib'
+
+export interface CircularAction<CircularActionContext> {
+  initialContext: () => CircularActionContext
+  currentContext: CircularActionContext
+  action: (
+    circularArray: Circular[],
+    context: CircularActionContext
+  ) => Promise<CircularActionContext>
```

Comment:
The callback probably shouldn't modify the context.```suggestion
  ) => Promise<void>
```